src/metabase/models/session.clj

```diff
@@ -71,21 +82,34 @@
   [:and
    [:map-of :keyword :any]
    [:map
-    [:id uuid?]
+    [:key string?]
     [:type [:enum :normal :full-app-embed]]]])
 
 (defmulti create-session!
   "Generate a new Session for a User. `session-type` is currently either `:password` (for email + password login) or
-  `:sso` (for other login types). Returns the newly generated Session."
+  `:sso` (for other login types). Returns the newly generated Session with the id as the plain-text session-key."
   {:arglists '([session-type user device-info])}
   (fn [session-type & _]
     session-type))
 
+(defn generate-session-key
+  "Generate a new session key."
+  []
+  (str (random-uuid)))
```

Comment:
Do we want session keys to still be UUIDs or would it be worth changing this?

Reply:
In general, I'd probably prefer to have them not be uuids. Something a bit longer and with more entropy could be nice. But mostly, a uuid may trigger some security researcher's "bad security smell" detectors since some uuid libraries are very much not random enough.

BUT: since we are comparing the key directly against the `id` column, we need a way to tell ids apart from keys, and the UUID format works well for that. So I was fine leaving it as a uuid.